server/resourcemanager/testutil/operations_list.go

```diff
@@ -1,109 +1,251 @@
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		expected := `{
+			"count": 0,
+			"items": []
+		}`
+
+		require.JSONEq(t, expected, jsonBody)
+	},
 }
 
-const OperationListInteralError Operation = "ListInteralError"
-
-type listInteralErrorOperation struct{}
+const OperationListSuccess Operation = "ListSuccess"
 
-func (op listInteralErrorOperation) postAssert(t *testing.T, ct contentType, rt ResourceTypeTest, testServer *httptest.Server) {
+var listSuccessOperation = operationTester{
+	name: OperationListSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		expected := `{
+			"count": 1,
+			"items": [` + ct.toJSON(rt.SampleJSON) + `]
+		}`
+
+		require.JSONEq(t, expected, jsonBody)
+	},
 }
 
-func (op listInteralErrorOperation) buildRequest(t *testing.T, testServer *httptest.Server, ct contentType, rt ResourceTypeTest) *http.Request {
-	return buildListRequest(
-		rt.ResourceType,
-		ct,
-		testServer,
-		t,
-	)
+const OperationListPaginatedAscendingSuccess Operation = "ListPaginatedAscendingSuccess"
+
+var listPaginatedAscendingSuccessOperation = operationTester{
+	name: OperationListPaginatedAscendingSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{
+				"take":          "2",
+				"skip":          "1",
+				"sortBy":        getSortFieldsString(rt.PaginationSortFields),
+				"sortDirection": "asc",
+			},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		var parsedJsonBody struct {
+			Count int              `json:count`
+			Items []map[string]any `json:items`
+		}
+		json.Unmarshal([]byte(jsonBody), &parsedJsonBody)
+
+		require.Equal(t, 3, parsedJsonBody.Count)
+
+		var prevVal any
+
+		sortFields := getSafeSortFields(rt.PaginationSortFields)
+
+		for _, field := range sortFields {
+			for _, item := range parsedJsonBody.Items {
+				if prevVal == nil {
+					prevVal = item[field]
+					continue
+				}
+				assert.True(t, compareFieldsLesserThan(prevVal, item[field]))
+
+				prevVal = item[field]
+			}
+		}
+	},
 }
 
-func (listInteralErrorOperation) name() Operation {
-	return OperationListInteralError
+const OperationListPaginatedDescendingSuccess Operation = "ListPaginatedDescendingSuccess"
+
+var listPaginatedDescendingSuccessOperation = operationTester{
+	name: OperationListPaginatedDescendingSuccess,
```

Comment:
I agree with that, but to do this change, we will need to change our operation structure to deal with more requests (today, it is fixed with only one request as part of the operation structure).

Can we unify this behavior in other PR? (to avoid increasing the scope of this PR)

Reply:
Never mind, after thinking for a while, our structure can have two `operationTester` with the same name. By doing that we could add both `asc` and `desc` in the same operation. 

It does make sense?